Mrj review ch6 (#401)
worldbank/d4di: chapters/data-analysis.tex

No modifications to data points are made at this stage, except for corrections of mistaken entries.

Cleaning is probably the most time-consuming of the stages discussed in this chapter.
You need to acquire an extensive understanding of the contents and structure of the raw data.
Explore the data set using tabulations, summaries, and descriptive plots.
Knowing your data set well will make it possible to do analysis.

\subsection{Identifying the identifier}

The first step in the cleaning process is to understand the level of observation in the data (what makes a row), 
and what variable or set of variables uniquely identifies each observations.
Ensuring that observations are uniquely and fully identified\sidenote{\url{https://dimewiki.worldbank.org/ID\_Variable\_Properties}}
is possibly the most important step in data cleaning.
It may be the case that the variable expected to be the unique identifier in fact is either incomplete or contains duplicates. 
This could be due to duplicate observations or errors in data entry.
It could also be the case that there is no identifying variable, or the identifier is a long string, such as a name.

> In this case cleaning begins by carefully creating a numeric variable that uniquely identifies the data. 

Reviewer: ```suggestion
In this case cleaning begins by carefully creating a variable that uniquely identifies the data. 
```

Does it have to be **numeric**? I previously though so, and I still prefer that, but we had a discussion about this once, and no one had good arguments to why "A12345", "8765YZ" would not be permitted. They are less memory efficient, but not to a degree where it has any impact whatsoever to the vast majority of projects.

As discussed in the previous chapter,
checking for duplicated entries is usually part of data quality monitoring,
and is ideally addressed as soon as data is received

> Note that while modern survey tools create unique observation identifiers,

Reviewer: ```suggestion
Note that while modern survey tools create unique identifiers for each submitted record,
```

The last step of data cleaning is to label and annotate the data,
so that all users have the information needed to interact with it.
There are three key steps: renaming, labeling and recoding. 
This is a key step to making the data easy to use, but it can be quite repetitive.
The \texttt{iecodebook} command suite, also part of \texttt{iefieldkit},
is designed to make some of the most tedious components of this process easier.\sidenote{
	\url{https://dimewiki.worldbank.org/iecodebook}}
\index{iecodebook}

First, \textbr{renaming}: for data with an accompanying survey instrument, 
it is useful to keep the same variable names in the cleaned dataset as in the survey instrument.
That way it's straightforward to link variables to the relevant survey question.
Second, \textbf{labeling}: applying labels makes it easier to understand your data as you explore it,
and thus reduces the risk of small errors making their way through into the analysis stage.
Variable and value labels should be accurate and concise.\sidenote{
	\url{https://dimewiki.worldbank.org/Data\_Cleaning\#Applying\_Labels}}

Third, \textbf{recoding}: codes for ``Don't know'', ``Refused to answer'', and

> other non-responses should be recoded into extended missing values.\sidenote{

Reviewer: "extended missing values" does not exist in all languages. It might even be a Stata terminology. Should we mention Stata here? We should somehow mention that it is important that we do not just delete these values. @luizaandrade , is there equivalent practice in R? 

We could say that the codes must be removed but records of them must still be kept. In Stata that can elegantly be done using extended missing values.

Reviewer 2: As far as I know, there are no extended missings in R. I have looked for it quite a bit. But I like Kris' suggestion.

Author: done in 946f782b993ca5188eba496b26f743f952497a90

Author: done in 946f782b993ca5188eba496b26f743f952497a90

In practice, however, as you analyze the data, 
different constructed variables will become necessary,
as well as subsets and other alterations to the data.
Even if construction and analysis are done concurrently, 
you should ways do the two in separate scripts. 
If every script that creates a table starts by loading a data set,
subsetting it, and manipulating variables,
any edits to construction need to be replicated in all scripts.
This increases the chances that at least one of them will have a different sample or variable definition.
Doing all variable construction in a single, separate script helps 
avoid this and ensure consistency across different outputs.

\subsection{Constructing analytical variables}
New variables created during the construction stage should be added to the data set, instead of overwriting the original information.
New variables should be assigned functional names. 
Ordering the data set so that related variables are together,
and adding notes to each of them as necessary will make your data set more user-friendly.

Before constructing new variables, 
you must check and double-check the value-assignments of questions,

> as well as the units and scales

Reviewer: ```suggestion
as well as the units and scales.
```
